src/httpserver.cpp

```diff
@@ -486,7 +478,10 @@ void StopHTTPServer()
     LogPrint(BCLog::HTTP, "Stopping HTTP server\n");
     if (workQueue) {
         LogPrint(BCLog::HTTP, "Waiting for HTTP worker threads to exit\n");
-        workQueue->WaitExit();
+        for (auto& thread: g_thread_http_workers) {
+            thread.join();
```

Comment:
From the [documentation](http://www.cplusplus.com/reference/thread/thread/~thread/), I don't think this is needed, `clear()` below is enough.

Reply:
No, if you delete a thread without either calling detach() or join(), terminate() is called. join() is not automatically called for you.